dune-precice/src/dune-perpendicular-flap.cc

```diff
@@ -0,0 +1,222 @@
+  operatorType stiffnessMatrix;
+  operatorAssembler.initialize(stiffnessMatrix);
+  Elastodynamics::StiffnessAssembler stiffnessAssembler(E, nu);
+  operatorAssembler.assemble(stiffnessAssembler, stiffnessMatrix, false);
+  
+  double rho = 3000.0;
+  diagonalType massMatrix(basis.size());
+  Elastodynamics::HRZLumpedMassAssembler massAssembler(rho);
+  operatorAssembler.assemble(massAssembler, massMatrix, true);
+  massMatrix.invert();
+
+  blockVector loadVector(basis.size()), displacementVector(basis.size()), velocityVector(basis.size()), accelerationVector(basis.size());
+  loadVector = 0.0, displacementVector = 0.0, velocityVector = 0.0, accelerationVector = 0.0;
+
+  // state quantatiy vector
+  std::vector<blockVector*> stateQuantaties;
+  stateQuantaties = {&displacementVector, &velocityVector, &accelerationVector};
+
+  // set dirichlet boundary
+  BitSetVector<dim> dirichletDofs(basis.size(), false);
+  auto dirichletPredicate = [](auto position) {return position[1] < 0.00001;};  
+  Functions::interpolate(basis, dirichletDofs, dirichletPredicate);
+ 
+  DiagonalMatrix<double, dim> I(1.0);
+  FieldMatrix<double, dim> OM(0.0);
+  
+  // set fixed Dirichlet entries in stiffnessmatrix
+  for( int i=0; i<stiffnessMatrix.N(); i++) {
+    if(dirichletDofs[i][0]) {
+      auto cIt = stiffnessMatrix[i].begin();
+      auto cEndIt = stiffnessMatrix[i].end();  
+      for( ; cIt!=cEndIt; ++cIt) {
+        *cIt = (i==cIt.index()) ? I : OM;
+      }
+    }
+  }
+  
+  // set fixed Dirichlet entries in massmatrix
+  for( int i=0; i<massMatrix.N(); i++) {
+    if(dirichletDofs[i][0]) {
+      auto cIt = massMatrix[i].begin();
+      auto cEndIt = massMatrix[i].end();  
+      for( ; cIt!=cEndIt; ++cIt) {
+        *cIt = (i==cIt.index()) ? I : OM;
+      }
+    }
+  }
+
+  FieldVector<double, dim> OV(0.0);
+
+  // set fixed Dirichlet entries in load vector
+  for( int i=0; i<loadVector.N(); i++) {
+    if(dirichletDofs[i][0]) {
+      loadVector[i] = OV;
+    }
+  }
+
+  // set neumann boundary
+  BlockVector<FieldVector<bool, dim>> neumannDofs(basis.size());
+  neumannDofs = false;
+  auto neumannPredicate = [](auto position) {return position[1] > -3.000001 || position[0] < 2.955555 || position[0] > 3.055555;};  
+  
+  Functions::interpolate(basis, neumannDofs, neumannPredicate);
+  
+  // generate global displacement function
+  using displacementRange = Dune::FieldVector<double, dim>;
+  auto displacementFunction = Functions::makeDiscreteGlobalBasisFunction<displacementRange> (basis, displacementVector);
+  
+  // generate output writer
+  double dt_out = 0.1;
+  auto vtkWriter = std::make_shared<SubsamplingVTKWriter<GridView>> (gridView, refinementLevels(0));
+  VTKSequenceWriter<GridView> vtkSequenceWriter(vtkWriter, "solid");
+  vtkWriter->addVertexData(displacementFunction, VTK::FieldInfo("displacement", VTK::FieldInfo::Type::vector, dim));
+  vtkSequenceWriter.write(0.0);
+
+  // setup looping
+  double t = 0.0;
+  double dt = 0.0001;
```

Comment:
Why do you need to select such a small time-step size? The tutorial uses actually `1e-2`?!

Reply:
Ah ok. This probably explains the deviation of the displacement amplitude as well (I can run a brief comparison tomorrow). You don't have any other time-stepping methods available, right? All in all, I can run the tutorial case andI will have a closer look at the code tomorrow.